### DISCUSSION TITLE
Unable to make queries to my server with the api

### DISCUSSION THREAD

**Discussion by Sean0203** [pufferpanel/pufferpanel#1209] [Q&A] [answered]
I'm trying to manipulate the API for PufferPanel in a python script. But it seems that after retrieving the access token I cannot get a basic request going. 
I always get error 401 (forbidden), so either the API is broken or I'm doing something wrong (probably the second option).

Here is the code I have at the moment, I hope someone can help me with this:

```python
import requests

# Setup
API_ENDPOINT = 'http://{SERVER IP}:8080/api'
CLIENT_ID = '{client id}'
CLIENT_SECRET = '{client secret}'
TOKEN_URL = 'http://{SERVER IP}:8080/oauth2/token'

request_data = {
        'grant_type': 'client_credentials',
        'client_id': CLIENT_ID,
        'client_secret': CLIENT_SECRET,
}

# retrieve access token
response = requests.post(TOKEN_URL, request_data)
access_token = response.json()['access_token']

# simple request to get node list
node_list_response = requests.get('http://{SERVER IP}:8080/api/nodes', headers={'Authorization': 'access_token ' + access_token})

# prints the response's code and content
print(node_list_response)
print(node_list_response.text)
```

**Comment by nepcore** [comment, marked as answer]
You're sending a header `Authorization: access_token sometoken` which is wrong, it needs to be `Authorization: Bearer sometoken`

**Reply by Sean0203** [reply]
Thanks for the quick reply! That was the issue.
